src/main/webapp/go.js

```diff
@@ -328,13 +335,16 @@ function updateRiddleToFinalMessage() {
 
 /**
  * Update the scavenger hunt data with the current destination that
- * the user is on.
+ * the user is on. Also sends back the ID of the scavenger hunt
+ * that the user is on.
  * @param {int} index Index of the current destination the user needs
  * to find.
  */
 function sendIndexToServlet(index) {
   const params = new URLSearchParams();
   params.append(INDEX_PARAM, index);
+  const pair = huntID.split('=');
+  params.append(pair[0], pair[1]);
```

Comment:
You always need to check values before you access them.  check if pair.length >=2